src/main/java/com/gluonhq/substrate/util/windows/MSIBundler.java

```diff
@@ -0,0 +1,325 @@
+    private final String sourceOS;
+    private final Path rootPath;
+
+    public MSIBundler(ProcessPaths paths, InternalProjectConfiguration projectConfiguration) {
+        this.paths = paths;
+        this.projectConfiguration = projectConfiguration;
+        this.sourceOS = projectConfiguration.getTargetTriplet().getOs();
+        this.rootPath = paths.getSourcePath().resolve(sourceOS);
+    }
+
+    public boolean createPackage(boolean sign) throws IOException, InterruptedException {
+
+        if (WixTool.CANDLE.getPath() == null || WixTool.LIGHT.getPath() == null) {
+            Logger.logSevere("Please make sure to install Wix Toolset (https://wixtoolset.org/) before proceeding.");
+            return false;
+        }
+
+        final String appName = projectConfiguration.getAppName();
+        final String version = Optional.ofNullable(projectConfiguration.getReleaseConfiguration().getVersionName()).orElse(DEFAULT_APP_VERSION);
+        Path localAppPath = paths.getAppPath().resolve(appName + ".exe");
+        if (!Files.exists(localAppPath)) {
+            throw new IOException("Error: " + appName + ".exe not found");
+        }
+        Logger.logInfo("Building exe for " + localAppPath);
+
+        /**
+         * Create directories and copy resources
+         */
+        Path tmpMSI = paths.getTmpPath().resolve("tmpMSI");
+        if (Files.exists(tmpMSI)) {
+            FileOps.deleteDirectory(tmpMSI);
+        }
+        Files.createDirectories(tmpMSI);
+
+        Path userAssets = rootPath.resolve(Constants.WIN_ASSETS_FOLDER);
+        // copy assets to gensrc/windows
+        Path windowsGenSrcPath = paths.getGenPath().resolve(sourceOS);
+        Path windowsAssetPath = windowsGenSrcPath.resolve(Constants.WIN_ASSETS_FOLDER);
+        Files.createDirectories(windowsAssetPath);
+        // Copy system resources and over-write it with user provided resources
+        FileOps.copyDirectoryFromResources("/native/windows/assets", windowsAssetPath);
+        Logger.logInfo("Default icon.ico image generated in " + windowsAssetPath + ".\n" +
+                "Consider copying it to " + rootPath + " before performing any modification");
+        if (Files.exists(userAssets)) {
+            FileOps.copyDirectory(userAssets, windowsAssetPath);
+        }
+
+        Path config = tmpMSI.resolve("config");
+        Files.createDirectories(config);
+        Path wixPath = config.resolve("main.wxs");
+        Path wixObjPath = wixPath.getParent().resolve(wixPath.getFileName() + ".wixobj");
+        Path msiPath = paths.getGvmPath().getParent().resolve(appName + "-" +  version  + ".msi");
+        FileOps.copyResource("/native/windows/wix/main.wxs", wixPath);
+        FileOps.copyDirectory(windowsAssetPath, config);
+
+        /**
+         * Wix Compile
+         */
+        Map<String, String> userInput = createAppDetailMap();
+        List<String> processArgs = new ArrayList<>(List.of(
+                WixTool.CANDLE.getPath(),
```

Comment:
The paths in this command need to be wrapped with quotes.

Reply:
I tried out this command and it doesn't fail for paths which contains space in it:

```
C:\Program Files (x86)\WiX Toolset v3.11\bin\candle.exe ...
```

The above path just worked fine.